src/interface/blas2_interface.hpp

```diff
@@ -551,14 +553,18 @@ typename sb_handle_t::event_t _spmv_impl(sb_handle_t& sb_handle, index_t _N,
   auto vx = make_vector_view(_vx, _incx, vector_size);
   auto vy = make_vector_view(_vy, _incy, vector_size);
 
-  auto spmv = make_spmv<local_range_x, local_range_y, uplo == uplo_type::Upper>(
-      _alpha, mA, vx, _beta, vy);
+  auto spmv =
+      make_xpmv<local_range_x, local_range_y, /*is_symmetric=*/true, is_upper,
+                /*is_transposed=*/false, /*is_unit=*/false>(_alpha, mA, vx,
```

Comment:
Thank you @pgorlani  for you work. I think these comments inside the template could be removed even if they increase readability.

Reply:
Thanks for your review, @s-Nick.

I prefer to keep these comments for having a clearer code.

Reply:
If the `make_xpmv` has a proper documentation, keeping this just create boiler plates and is not necessary.

Reply:
Hi @mehdi-goli,

I think that `make_xpmv` is self-explanatory due to the template parameter names and types. This is just my opinion, I'm open to suggestions. That is the reason I think that recalling the template parameters here can be helpful considering also the fact that `make_xpmv` is defined on another file.

I don't think these three little comments will bloat the code, but I can remove them if you think so.

Reply:
In SYCL-BLAS we provide doxygen for all the functions and classes we generate. So we must follow that. and one place will capture all these info regarding the function declaration. It is not only this function, This is for the repetition of such functions so imagine everywhere such similar functions used, we put multiple comments there. So if we have proper doxygen for the function, the generator provide proper documentation. hence, no need for such comments

Reply:
https://github.com/codeplaysoftware/sycl-blas/pull/417/commits/4dff277d2af77538c07f40e4dad3ad681f1c704a updates the documentation.